bluetooth_mesh/mesh.py

```diff
@@ -147,6 +147,52 @@ def transport_pdu(self, application_key, seq, iv_index, szmic=False):
                                       seg, akf, aid, upper_transport_pdu).bytes
 
 
+class ControlMessage:
+    MAX_TRANSPORT_PDU = 15
+    SEGMENT_SIZE = 12
+
+    def __init__(self, src, dst, ttl, opcode, payload):
+        self.src = src
+        self.dst = dst
+        self.ttl = ttl
+        self.ctl = True
+        self.opcode = opcode
+        self.payload = payload
+
+        self.nonce = Nonce(self.src, self.dst, self.ttl, self.ctl)
+
+    def transport_pdu(self, application_key, seq, iv_index):
+
+        seg = len(self.payload) > self.MAX_TRANSPORT_PDU
+        seq_zero = seq & 0x1fff
+
+        if seg:
+            segments = list(self.payload[i:i + self.SEGMENT_SIZE]
+                            for i in range(0, len(self.payload), self.SEGMENT_SIZE))
+
+            seg_n = len(segments) - 1
+
+            for seg_o, segment in enumerate(segments):
+                yield seq + seg_o, bitstring.pack('uint:1, uint:7, pad:1, uint:13, uint:5, uint:5, bytes',
+                                                  seg, self.opcode, seq_zero, seg_o, seg_n, segment).bytes
+        else:
+            yield seq, bitstring.pack('uint:1, uint:7, bytes',
+                                      seg, self.opcode, self.payload).bytes
+
+
+class SegmentAckMessage(ControlMessage):
+
+    def __init__(self, src, dst, ttl, seq, ack_segments, obo=False):
+        self.obo = obo
+        self.seq = seq
```

Comment:
`seq` means a different thing than `seg_zero` - the former is just a sequence number, while the latter denotes SAR transaction number. Please don't confuse them.